akka-persistence/src/main/scala/akka/persistence/Persistence.scala

```diff
@@ -82,22 +83,45 @@ trait PersistenceIdentity {
   def persistenceId: String
 
   /**
-   * Configuration id of the journal plugin servicing this persistent actor or view.
+   * Configuration id of the journal plugin servicing this persistent actor.
    * When empty, looks in `akka.persistence.journal.plugin` to find configuration entry path.
    * When configured, uses `journalPluginId` as absolute path to the journal configuration entry.
    * Configuration entry must contain few required fields, such as `class`. See `src/main/resources/reference.conf`.
    */
   def journalPluginId: String = ""
 
   /**
-   * Configuration id of the snapshot plugin servicing this persistent actor or view.
+   * Configuration id of the snapshot plugin servicing this persistent actor.
    * When empty, looks in `akka.persistence.snapshot-store.plugin` to find configuration entry path.
    * When configured, uses `snapshotPluginId` as absolute path to the snapshot store configuration entry.
    * Configuration entry must contain few required fields, such as `class`. See `src/main/resources/reference.conf`.
    */
   def snapshotPluginId: String = ""
 
+  /**
+   * Additional configuration of the journal plugin servicing this persistent actor.
+   * When empty, the whole configuration of the journal plugin will be taken from the [[Config]] loaded into the
+   * [[ActorSystem]].
+   * When configured, the journal plugin configuration will be taken from this [[Config]] merged with the [[Config]]
+   * loaded into the [[ActorSystem]].
+   *
+   * @return an additional configuration used to configure the journal plugin.
+   */
+  def journalPluginConfig: Config = ConfigFactory.empty
+
+  /**
+   * Additional configuration of the snapshot plugin servicing this persistent actor.
+   * When empty, the whole configuration of the snapshot plugin will be taken from the [[Config]] loaded into the
+   * [[ActorSystem]].
+   * When configured, the snapshot plugin configuration will be taken from this [[Config]] merged with the [[Config]]
+   * loaded into the [[ActorSystem]].
+   *
+   * @return an additional configuration used to configure the snapshot plugin.
+   */
+  def snapshotPluginConfig: Config = ConfigFactory.empty
```

Comment:
Adding those methods is not binary compatible, but you can add a new trait with these and pattern match on that when doing `extension.journalFor` in Eventsourced.scala.

Reply:
Good point, we will see to take it into account!

Reply:
No worries about adding a new trait including the 2 methods we initially added in `PersistenceIdentity`. Unsure about the naming but for discussion purposes let's name this trait `ProvidedPluginConfig`.

Then, since the `EventSourced` trait seems to be part of the internal API, does that mean that we could mix this new `ProvidedPluginConfig` trait and ignore binary incompatibility at this point ?

Reply:
Well actually that doesn't change anything since `PersistentActor` implements the `EventSourced` trait, so the binary compatibility issue is still around.

@patriknw : could you please elaborate a bit about the fix you had in mind ?

Reply:
@pradasouvanlasy sorry for delay, this is what I mean:

Instead of defining those methods here you should define a new interface:
```
trait RuntimePluginConfig {
  def journalPluginConfig: Config
  def snapshotPluginConfig: Config
}
```

Then:
```
private[persistence] lazy val journal = {
  val journalPluginConfig = this match {
    case c: journalPluginConfig => c.journalPluginConfig
    case _ => ConfigFactory.empty
  }
  extension.journalFor(journalPluginId, journalPluginConfig)
}
```
similar for `snapshotStore`.

User must then:
```
class MyPersistentActor extends PersistentActor with RuntimePluginConfig {
  override def journalPluginConfig...
}
```